lib/src/backend/mod.rs

```diff
@@ -0,0 +1,57 @@
+use anyhow::{anyhow, Result};
+use clap::ValueEnum;
+
+use ostree::glib;
+use ostree_ext::container::OstreeImageReference;
+use ostree_ext::keyfileext::KeyFileExt;
+use ostree_ext::ostree;
+use ostree_ext::sysroot::SysrootLock;
+
+use crate::spec::ImageStatus;
+
+mod ostree_container;
+
+pub(crate) trait Backend {
+    fn backend(&self) -> Result<Box<dyn BackendImpl>>;
+}
+
+pub(crate) trait BackendImpl {
+    fn imagestatus(
+        &self,
+        sysroot: &SysrootLock,
+        deployment: &ostree::Deployment,
+        image: OstreeImageReference,
+    ) -> Result<(Option<ImageStatus>, Option<ImageStatus>)>;
```

Comment:
Not a big deal but we probably want a struct here for the return value, just looking at it it's not clear that one is the current and one is the cached status. (Can we directly return the spec value?)

Reply:
I've added a `backend::CachedImageStatus` struct to wrap this.  Re: spec value, I didn't want to directly return a full `BootEntry` since it contains ostree ("frontend"?) specific things that aren't relevant to the backend.  Felt like a clean break that the backend only cares about the image/cached_update, and the ostree parts are shared for `incompatible`, `pinned`, and `ostree`.